csrc/scheduler/vectorize_helper.cpp

```diff
@@ -842,6 +842,59 @@ std::vector<std::unordered_map<TensorView*, Val*>> getTvToContigInnerSizeMapsOf(
   return mappers;
 }
 
+// Check if a traversal from vectorized reference IDs may reach the
+// IDs of a resize expr without visiting the Resize expr itself. That's
+// problematic for the vectorization analysis as the spanning-tree
+// based analysis may miss the constraint by the Resize expr.
+//
+// For this analysis, we start a traversal from the vectorized
+// reference IDs to both the input and output of the Resize expr but
+// disallow visiting the Resize expr itself. If the traversal is still
+// successful, it means there's a path from the reference IDs to the
+// resize input and output IDs without visiting the Resize expr.
+//
+// Permissive BFS is used in this traversal as the vectorized
+// reference IDs may not have all the dependencies for the
+// traversal. For example, suppose there's a split resshape, and only
+// the innermost ID is vectorized. The standard BFS is not able to
+// move forward if only the vectorized ID is give as the backward
+// split requires both outputs to be presented.
+class CanSkipResize : public ValGraphPermissiveBFS {
+ public:
+  static bool run(
+      const ValGraph& graph,
+      const ValGroups& ref_groups,
+      Resize* resize) {
+    ValGroups resize_in_out_groups;
+    resize_in_out_groups.pushBack(graph.toGroup(resize->in()));
+    resize_in_out_groups.pushBack(graph.toGroup(resize->out()));
+    CanSkipResize bfs(graph, ref_groups, resize_in_out_groups, resize);
+    bfs.traverse();
+    return bfs.allToNodesVisited();
```

Comment:
qq: here we are calling `allToNodesVisited()`? but the init function below has `/*require_all_to_visited=*/false,`, so we are returning true here as long as a single node is visited in the target, which I think is the right behavior.

But the function name is somewhat confusing.

Reply:
No, the traversal should continue until no further progress is made. The `require_all_to_visited` flag means it's considered an error if not all of the `to` nodes were not able to reach.

Here, we just want to check all of the `to` nodes are reachable. It isn't an error even if not.

Reply:
> so we are returning true here as long as a single node is visited in the target, which I think is the right behavior.

sorry I got confused myself. This returns true indicating it's safe to skip the check. So `allToNodesVisited` is the proper name for the function.

Thanks for elaborating on this one.